repository/repo-operations.go

```diff
@@ -442,16 +455,74 @@ func openPullRequest(config *config.GitXargsConfig, repo *github.Repository, bra
 	// Configure pull request options that the GitHub client accepts when making calls to open new pull requests
 	newPR := &github.NewPullRequest{
 		Title:               github.String(titleToUse),
-		Head:                github.String(branch),
+		Head:                github.String(pr.Branch),
 		Base:                github.String(repoDefaultBranch),
 		Body:                github.String(descriptionToUse),
 		MaintainerCanModify: github.Bool(true),
 		Draft:               github.Bool(config.Draft),
 	}
 
 	// Make a pull request via the Github API
-	pr, resp, err := config.GithubClient.PullRequests.Create(context.Background(), *repo.GetOwner().Login, repo.GetName(), newPR)
+	githubPR, resp, err := config.GithubClient.PullRequests.Create(context.Background(), *pr.Repo.GetOwner().Login, pr.Repo.GetName(), newPR)
+
+	// the go-github library's CheckResponse method can return two different types of rate limiting error:
+	// 1. AbuseRateLimitError which may contain a Retry-After header whose value we can use to slow down, or
+	// 2. RateLimitError which may contain information about when the rate limit will be removed, that we can also use to slow down
+	// Therefore, we need to use type assertions to test for each type of error response, and accordingly fetch the data it may contain
+	// about how long git-xargs should wait before its next attempt to open a pull request
+	githubErr := github.CheckResponse(resp.Response)
+
+	if githubErr != nil {
```

Comment:
Can take care of this in a followup as well. It's hard to understand what's going on because this block is quite large. Basically it's adding a delay, or it's returning `nil` if rate limited. Is there a case for pulling this whole thing out into a separate function? That way it's also clearer that `err` that you captured in the pull request API call is being handled, just further down.